src/common/reactNativeProjectHelper.ts

```diff
@@ -13,8 +15,38 @@ export class ReactNativeProjectHelper {
         return ["0.54.0", "0.54.1", "0.54.2", "0.54.3", "0.54.4"];
     }
 
-    public static getReactNativeVersion(projectRoot: string) {
-        return new CommandExecutor(projectRoot).getReactNativeVersion();
+    public static getReactNativeVersion(projectRoot: string): Q.Promise<string> {
+        return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(projectRoot)
+            .catch(err => {
+                return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(projectRoot);
```

Comment:
What happens if there is not 'react-native' in dev dependencies or dependencies?

Reply:
If neither `dependencies` nor `devDependencies` contain `react-native` field, then an empty string will be returned. If some error occurs while getting React Native version, empty string will be returned too.